app/models/problem.rb

```diff
@@ -207,10 +207,17 @@ def unmerge!
 
     # recache each new problem
     new_problems.each(&:recache)
-
     new_problems
   end
 
+  # Returns an array of unix epoc timestamps for the created_at field of
+  # all notices sincethe since parameter
```

Comment:
Does this go fetch all zillion notices, put them in memory and then map them all right here? Or does this syntax correspond to some mongo optimized map-reduce thing? (https://docs.mongodb.com/manual/core/map-reduce/)

I haven't tried it, but if you have a literal million notices, this is likely to be prohibitively slow and tie up puma workers doing a lot of iterating and mapping and such. I *think* there's probably an optimal way to deal with this and still get a usable report from mongo even with millions of notices, but I'm not 100% sure. Any thoughts?

Reply:
yes, this was not extremely straightforward, but ultimately a good suggestion.  it's about 20 times faster for one data set using mongo aggregations. :)